bin/check_samplesheet_create_tables.py

```diff
@@ -0,0 +1,271 @@
+
+    parser = argparse.ArgumentParser(description=description, epilog=epilog)
+    parser.add_argument(
+        "-i",
+        "--input",
+        required=True,
+        metavar="FILE",
+        type=argparse.FileType("r"),
+        help="Input samplesheet file containing: condition, type, microbiome_path, alleles, weights_path.",
+    )
+    parser.add_argument(
+        "-m",
+        "--microbiomes",
+        required=True,
+        metavar="FILE",
+        type=argparse.FileType("w"),
+        help="Output file containing: microbiome_id, microbiome_path, microbiome_type, weights_path.",
+    )
+    parser.add_argument(
+        "-c",
+        "--conditions",
+        required=True,
+        metavar="FILE",
+        type=argparse.FileType("w"),
+        help="Output file containing: condition_id, condition_name, microbiome_id.",
+    )
+    parser.add_argument(
+        "-a",
+        "--alleles",
+        required=True,
+        metavar="FILE",
+        type=argparse.FileType("w"),
+        help="Output file containing: allele_id, allele_name.",
+    )
+    parser.add_argument(
+        "-ca",
+        "--conditions_alleles",
+        required=True,
+        metavar="FILE",
+        type=argparse.FileType("w"),
+        help="Output file containing: condition_id, allele_id.",
+    )
+    parser.add_argument(
+        "-pm",
+        "--prediction_method",
+        required=True,
+        metavar="STRING",
+        type=str,
+        help="Chosen method for epitope prediction",
+    )
+    parser.add_argument(
+        "-pmv",
+        "--pred_method_version",
+        required=True,
+        metavar="STRING",
+        type=str,
+        help="Chosen version of epitope prediction method",
+    )
+    parser.add_argument(
+        "-pl",
+        "--peptide_lengths",
+        required=True,
+        metavar="TUPLE",
+        nargs="+",
+        help="Peptide lengths as given in parameters (min max)",
+    )
+    return parser.parse_args(args)
+
+
+def print_error(error, context="Line", context_str=""):
+    error_str = f"ERROR: Please check samplesheet -> {error}"
+    if context != "" and context_str != "":
+        error_str = f"ERROR: Please check samplesheet -> {error}\n{context.strip()}: '{context_str.strip()}'"
+    print(error_str)
+    sys.exit(1)
+
+
+def process_samplesheet(args):
+    """
+    Check that the tabular samplesheet has the structure expected by nf-core/metapep and create the data tables.
+
+    Header structure:
+    condition,type,microbiome_path,alleles,weights_path
+
+    For an example see:
+    https://github.com/nf-core/metapep/raw/dev/assets/samplesheet.csv
+    """
+    input_table = pd.read_csv(args.input)
+    input_table_cp = input_table.copy()
+
+    # check if microbiome_path file extensions are valid
```

Comment:
Are these samplesheet checks needed or can you outsource (most of) them in the assets/schema_input.json and let nf-validation take care of that?

Reply:
The problem to use the schema is that each row needs to fit specific requirements, for the chosen input type: e.g. `taxa` input needs `.tsv` files while assembly needs .fa(sta).gz